packages/core/src/app/customer/Customer.tsx

```diff
@@ -433,11 +435,23 @@ class Customer extends Component<CustomerProps & WithCheckoutCustomerProps & Ana
     };
 
     private handleCreateAccount: (values: CreateAccountFormValues) => void = async (values) => {
-        const { createAccount = noop, onAccountCreated = noop } = this.props;
+        const {
+            executePaymentMethodCheckout,
+            createAccount = noop,
+            onAccountCreated = noop,
+            providerWithCustomCheckout,
+        } = this.props;
 
         await createAccount(mapCreateAccountFromFormValues(values));
 
-        onAccountCreated();
+        if (providerWithCustomCheckout === PaymentMethodId.Braintree || providerWithCustomCheckout === PaymentMethodId.BraintreeAcceleratedCheckout) {
```

Comment:
I have seen this check in quite a few places now, can we move it as a util method and use a method in places?